src/app/core/components/language-mismatch/language-mismatch.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { mapPending } from 'src/app/shared/operators/map-pending';
+import { UserSessionService } from 'src/app/shared/services/user-session.service';
+import { LocaleService } from '../../services/localeService';
+
+@Component({
+  selector: 'alg-language-mismatch',
+  templateUrl: './language-mismatch.component.html',
+  styleUrls: [ './language-mismatch.component.scss' ],
+})
+export class LanguageMismatchComponent {
+
+  readonly currentLanguage = this.localeService.currentLang?.tag;
+  readonly languages = this.localeService.languages;
+
+  languageMismatch$ = this.sessionService.userProfile$.pipe(
+    map(profile => {
+      const hasLanguageMismatch = !!this.currentLanguage && profile.defaultLanguage !== this.currentLanguage;
+      return hasLanguageMismatch ? {
+        userDefaultLanguage: profile.defaultLanguage,
+        userDefaultLanguageIsSupported: this.languages.some(({ tag }) => tag === profile.defaultLanguage),
+      } : undefined;
+    }),
+  )
+  updating$?: Observable<boolean>
```

Comment:
I'm a bit annoyed by the fact this variable is optional and not read-only. We are a bit playing with the limits of Angular template engine to know how `*ngIf="updating$ | async"` works when it is not set, when it is set once, when it is replaced by another observable.  🤔 I have to think about this.

Reply:
Fixed